guides/release/components/template-lifecycle-dom-and-modifiers.md

```diff
@@ -340,40 +349,71 @@ export default class AudioPlayerComponent extends Component {
   play() {
     // TODO
   }
-}
-```
-
-We'd like to call the `play` method on our `<audio>` element, but how do we get access to that element?
-
-We can give our component access to elements inside of it by using [`ember-ref-modifier`](https://github.com/lifeart/ember-ref-modifier).
-
-> The `{{ref}}` modifier takes a property name and an object, and assigns the modifier's element to that property on the object.
-
-In this case, we'll assign the `<audio>` element to the `audioElement` property in our component:
-
-```handlebars {data-filename="app/components/audio-player.hbs" data-diff="-1,+2"}
-<audio src={{@srcURL}} />
-<audio src={{@srcURL}} {{ref this "audioElement"}} />
 
-<button type="button" {{on "click" this.play}}>Play</button>
-<button type="button">Pause</button>
+  @action
+  pause() {
+    // TODO
+  }
+}
 ```
 
-Now, the component can access the audio element in the `play` method:
+From the component's point of view, all it needs to do is track whether it should be playing or not, since we're going to manage the DOM element with the modifier. So we can just use `@tracked` and a normal property here.
 
-```js {data-filename="app/components/audio-player.js" data-diff="-7,+8"}
+```js {data-filename="app/components/audio-player.js" data-diff="+2,+6,+7,-10,+11,-16,+17"}
 import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
 import { action } from "@ember/object";
 
 export default class AudioPlayerComponent extends Component {
+  @tracked isPlaying = false;
+
   @action
   play() {
     // TODO
-    this.audioElement.play();
+    this.isPlaying = true;
+  }
+
+  @action
+  pause() {
+    // TODO
+    this.isPlaying = false;
   }
 }
 ```
 
+Now the component is managing the state of the player, but we still need to translate that into the DOM method calls. So let's create a modifier!
+
+```bash
+ember install ember-modifier
+ember generate modifier play-when
+```
+
+In this modifier, we'll simply take an argument that specifies whether we should be playing or paused.
+
+```js {data-filename="app/modifiers/play-when.js"}
+import { modifier } from "ember-modifier";
+
+export default modifier((element, [isPlaying]) => {
+  if (isPlaying) {
+    element.play();
+  } else {
+    element.pause();
+  }
+});
+```
+
+We can now attach the modifier to the `audio` element:
+
+```handlebars {data-filename="app/components/audio-player.hbs" data-diff="-1,+2"}
+<audio src={{@srcURL}} />
+<audio src={{@srcURL}} {{play-when this.isPlaying}} />
+
+<button type="button" {{on "click" this.play}}>Play</button>
```

Comment:
wondering, is it real use-case? (looks like we could remove `play-when` modifier and handle play logic in actions at all)

Reply:
usecase from our project:

```js
export default modifier(function scrollTop(element, [enabled]) {
  if (enabled) {
    element.scrollIntoView({behavior: "smooth", block: "start", inline: "nearest"});
  }
});
```

Reply:
or we could add more logic to `play-when`, to toggle `isPlaying` once audio will be finished (to get it close to real-life)

Reply:
A couple notes here—

1. This is in fact a real-world use case (as well as the example used originally here). The implementation I put up is *very* similar to one we designed for use in our own app.
2. My take on maintainability is a little different, I think: maintaining a very simple modifier and its tests is actually a big *win*, because it only has one thing it does and can focus on doing it well.